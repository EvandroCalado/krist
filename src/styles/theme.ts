export default {
  colors: {
    primary: '#1d4ed8',
    secondary: '#dc2626',
    warning: '#e11d48',
    success: '#16a34a',
    info: '#3b82f6',
    white: '#ffffff',
    black: '#000000',
    background: '#f5f5f5',
    text: '#262626',
  },
  font: {
    family: {
      primary: "'Inter', sans-serif;",
    },
    sizes: {
      xs: '.8rem',
      sm: '1.4rem',
      md: '2.4rem',
      lg: '3.2rem',
      xl: '4rem',
      '2xl': '4.8rem',
      '3xl': '5.6rem',
      '4xl': '6.4rem',
      '5xl': '7.2rem',
      '6xl': '8.0rem',
      '7xl': '9.6rem',
    },
  },
  media: {
    lteOrEqSmallest: '(max-width: 320px)',
    lteOrEqSmall: '(max-width: 576px)',
    lteOrEqMedium: '(max-width: 768px)',
    lteOrEqLarge: '(max-width: 992px)',
    gteOrEqSmallest: '(min-width: 320px)',
    gteOrEqSmall: '(min-width: 576px)',
    gteOrEqMedium: '(min-width: 768px)',
    gteOrEqLarge: '(min-width: 992px)',
  },
  spacings: {
    xs: '0.5rem',
    sm: '1rem',
    md: '2rem',
    lg: '3rem',
    xl: '4rem',
    '2xl': '5rem',
    '3xl': '6rem',
    '4xl': '7rem',
    '5xl': '8rem',
    '6xl': '9rem',
    '7xl': '10rem',
  },
  frameSizes: {
    xs: '8rem',
    sm: '12rem',
    md: '24rem',
    lg: '32rem',
    xl: '40rem',
    '2xl': '56rem',
    '3xl': '64rem',
    '4xl': '80rem',
    '5xl': '104rem',
    '6xl': '120rem',
    '7xl': '134rem',
  },
  transitions: {
    fastest: 'all 100ms ease-in-out',
    faster: 'all 200ms ease-in-out',
    fast: 'all 300ms ease-in-out',
    normal: 'all 500ms ease-in-out',
    slow: 'all 600ms ease-in-out',
    slower: 'all 800ms ease-in-out',
    slowest: 'all 1s ease-in-out',
  },
  layers: {
    normal: 0,
    layer1: 1,
    layer2: 2,
    layer3: 3,
    layer4: 4,
    layer5: 5,
    layer6: 6,
    layer7: 7,
    layer8: 8,
    layer9: 9,
    aboveAll: 10,
  },
} as const;
