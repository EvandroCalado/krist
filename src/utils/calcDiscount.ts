export const calcDiscount = (value: number, discount: number) => {
  const valueWithDiscount = (value * (100 - discount)) / 100;

  if (discount === 0) return value;

  return valueWithDiscount.toFixed(2);
};
