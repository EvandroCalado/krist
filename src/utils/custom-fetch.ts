import axios from 'axios';

export const customFetch = axios.create({
  baseURL: import.meta.env.VITE_STRAPI_URL,
});
