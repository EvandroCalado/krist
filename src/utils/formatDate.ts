export const formatDate = (dateString: string): string => {
  const date = new Date(dateString);

  return date.toLocaleString('pt-BR', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });
};
