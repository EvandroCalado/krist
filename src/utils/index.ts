export * from './calcDiscount';
export * from './custom-fetch';
export * from './formatDate';
export * from './linksProfileMenu';
export * from './test-utils';
