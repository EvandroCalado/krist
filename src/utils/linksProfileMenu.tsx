import { Heart, MapPin, Package, Settings, UserRound } from 'lucide-react';

export const linksProfileMenu = [
  {
    id: 1,
    label: 'informações pessoais',
    path: 'profile',
    icon: <UserRound size={16} />,
  },
  {
    id: 2,
    label: 'meus pedidos',
    path: 'orders',
    icon: <Package size={16} />,
  },
  {
    id: 3,
    label: 'lista de desejos',
    path: 'wishlist',
    icon: <Heart size={16} />,
  },
  {
    id: 4,
    label: 'meus endereços',
    path: 'addresses',
    icon: <MapPin size={16} />,
  },
  {
    id: 5,
    label: 'configurar',
    path: 'settings',
    icon: <Settings size={16} />,
  },
];
