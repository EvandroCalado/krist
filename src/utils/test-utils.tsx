import { render, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from 'styles/theme';

type CustomRenderProps = Omit<RenderOptions, 'queries'>;

const customRender = (
  ui: ReactElement,
  { ...renderOptions }: CustomRenderProps = {},
) => render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions);

export * from '@testing-library/react';
export { customRender as render };
