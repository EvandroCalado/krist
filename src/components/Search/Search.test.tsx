import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { customRender } from 'utils/test-utils';

import theme from 'styles/theme';

import { Search } from '.';

describe('Search', () => {
  it('should render default', () => {
    customRender(<Search />);

    const search = screen.getByRole('searchbox');

    expect(search).toBeInTheDocument();
    expect(search).toHaveAttribute('name', 'q');
    expect(search).toHaveValue('');
  });

  it('should render with defaultValue', () => {
    customRender(<Search defaultValue="test" />);

    const search = screen.getByRole('searchbox');

    expect(search).toHaveValue('test');
  });

  it('should render with send attribute', () => {
    customRender(<Search placeholder="test" />);

    const search = screen.getByRole('searchbox');

    expect(search).toHaveAttribute('placeholder', 'test');
  });

  it('should render with query value', () => {
    const route = '/?q=query';

    render(
      <MemoryRouter initialEntries={[route]}>
        <ThemeProvider theme={theme}>
          <Search />
        </ThemeProvider>
      </MemoryRouter>,
    );

    const search = screen.getByRole('searchbox');

    expect(search).toHaveValue('query');
  });

  it('should render a snapshot', () => {
    const { container } = customRender(<Search />);

    expect(container).toMatchSnapshot();
  });
});
