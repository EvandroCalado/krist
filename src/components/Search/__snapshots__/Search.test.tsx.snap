// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render a snapshot 1`] = `
.c0 {
  position: relative;
  display: flex;
  align-items: center;
}

.c0>input {
  width: 100%;
  outline: none;
  border: 1px solid #131118;
  padding: 1rem 4rem 1rem 1.6rem;
  border-radius: .8rem;
}

.c0>input:focus {
  outline: 3px solid #131118;
  outline-offset: 1px;
}

.c0>svg {
  position: absolute;
  right: 1.6rem;
}

<div>
  <div
    class="c0 search"
  >
    <input
      name="q"
      type="search"
      value=""
    />
    <svg
      class="lucide lucide-search "
      fill="none"
      height="18"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="11"
        cy="11"
        r="8"
      />
      <path
        d="m21 21-4.3-4.3"
      />
    </svg>
  </div>
  ,
</div>
`;
